frontend/src/scenes/session-recordings/player/modal/SessionPlayerModal.tsx

```diff
@@ -1,11 +1,24 @@
 import { LemonModal } from '@posthog/lemon-ui'
 import { BindLogic, useActions, useValues } from 'kea'
+import { useEffect, useState } from 'react'
 import { SessionRecordingPlayer } from 'scenes/session-recordings/player/SessionRecordingPlayer'
 
 import { PlayerMeta } from '../PlayerMeta'
 import { sessionRecordingPlayerLogic, SessionRecordingPlayerLogicProps } from '../sessionRecordingPlayerLogic'
 import { sessionPlayerModalLogic } from './sessionPlayerModalLogic'
 
+let sessionPlayerModalSingletonRendered = false
+
+/**
+ * When SessionPlayerModal is present in the page you can call `openSessionPlayer` action to open the modal
+ * and play a given session
+ *
+ * SessionPlayerModal is templated into the page by multiple components
+ * It has to be present _once_ for the player modal to work
+ * But gets very unhappy if there are multiple instances
+ * So, it is written as a singleton that attempts to only render once
+ *
+ */
```

Comment:
Please ignore my above comment 😂 I should have read the entire PR first

Reply:
no, you're right, it's equally messy now :)